src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -195,90 +114,114 @@ const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
-            context_module: ContextModule.ordersShipping,
-            context_page_owner_type: OwnerType.ordersShipping,
-            context_page_owner_id: contextPageOwnerSlug,
-            user_id: userId,
-            flow: "user adding shipping address",
-            subject: "Check your delivery address",
-            option: "review and confirm",
-          })
-        }
+        addressOptions = [inputOption]
+        verificationPath = VerificationPath.REVIEW_AND_CONFIRM
       }
     }
-  }, [
-    inputAddress,
-    onChosenAddress,
-    suggestedAddresses,
-    verificationStatus,
-    trackEvent,
-    userId,
-    contextPageOwnerSlug,
-    hasError,
-    verificationInput,
-  ])
-
-  if (verificationStatus === "VERIFIED_NO_CHANGE" || hasError)
-    return <div data-testid="emptyAddressVerification"></div>
+  }
 
-  if (!modalType || addressOptions.length === 0) return null
+  const modalTitle = modalTitles[verificationPath]
 
-  return (
+  const [selectedAddressKey, setSelectedAddressKey] = useState<
+    AddressOptionKey | undefined
+  >(addressOptions[0]?.key)
+
+  // perform only once when the flow first loads
+  useEffect(() => {
+    if (verificationPath === VerificationPath.ERROR_IMMEDIATE_CONFIRM) {
+      const fallbackOption = fallbackFromFormValues(verificationInput)
+      onChosenAddress(AddressVerifiedBy.USER, fallbackOption.address)
+    } else if (verificationPath === VerificationPath.IMMEDIATE_CONFIRM) {
+      onChosenAddress(AddressVerifiedBy.ARTSY, addressOptions[0].address!)
+    } else {
+      setShowModal(true)
+
+      trackViewedModal({
+        subject: modalTitle!,
+      })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const chooseAddress = useCallback(() => {
+    if (!selectedAddressKey) return
+
+    const selectedAddress = addressOptions.find(
+      option => option.key === selectedAddressKey
+    )
+
+    const verifiedBy =
+      selectedAddressKey === USER_INPUT
+        ? AddressVerifiedBy.USER
+        : AddressVerifiedBy.ARTSY
+
+    if (selectedAddress) {
+      setShowModal(false)
+      onChosenAddress(verifiedBy, selectedAddress.address)
+    }
+  }, [addressOptions, onChosenAddress, selectedAddressKey])
+
+  const handleCloseModal = ({
+    label,
+    subject,
+  }: {
+    label: string
+    subject: string
+  }) => {
+    trackClosedModal({
+      subject,
+      label,
+    })
+    onClose()
+  }
+
+  if (
+    [
+      VerificationPath.ERROR_IMMEDIATE_CONFIRM,
+      VerificationPath.IMMEDIATE_CONFIRM,
+    ].includes(verificationPath)
+  ) {
+    return <div data-testid="emptyAddressVerification"></div>
+  }
+
+  return !showModal ? null : (
     <ModalDialog
       width={["100%", 590]}
       height={["100vh", "auto"]}
       m={[0, 2]}
-      title={modalTitles[modalType]}
-      onClose={handleClose}
+      title={modalTitle!}
+      onClose={() => {
+        handleCloseModal({ label: "close modal", subject: modalTitle! })
```

Comment:
If I remember correctly, the label should be `null` when the modal is closed via clicking "X", clicking away, or hitting <ESC>. "close modal" is not a valid label from the [spreadsheet](https://docs.google.com/spreadsheets/d/1c65vugkF2C4YuXwBXno8aqsBh4B2fTwxiLmgY_MhjWc/edit#gid=0).

Reply:
Ah, i was looking into the palette docs and couldn't find a handler for clicking outside the modal so i decided to just put this for now. I also ran into an issue with some of these (like `option` where the cohesion scheme didn't permit null even though it was the specified value.

Reply:
resolved in latest commit [9bb308f](https://github.com/artsy/force/pull/12778/commits/9bb308f0717dc996fc0314135e65404ad56d3a85) after working with dayane.